contracts/proposal_fix.js

```diff
@@ -0,0 +1,250 @@
+    cAaveStrategyProxy.address
+  );
+
+  // Governance proposal
+  return {
+    name: "Deploy all new contracts and migrate all funds",
+    actions: [
+      {
+        // Claim governance of OUSDProxy, transferred in 020_new_governor
+        contract: cOUSDProxy,
+        signature: "claimGovernance()",
+      },
+      {
+        // Claim governance of cVaultProxy, transferred in 020_new_governor
+        contract: cVaultProxy,
+        signature: "claimGovernance()",
+      },
+      {
+        // Claim governance of Buyback
+        contract: cBuyback,
+        signature: "claimGovernance()",
+      },
+      {
+        // Claim governance of old Buyback
+        contract: cOldBuyback,
+        signature: "claimGovernance()",
+      },
+      {
+        // Claim Governance of Compound strategy
+        contract: cCompoundStrategy,
+        signature: "claimGovernance()",
+      },
+      {
+        // Claim Governance of Aave strategy
+        contract: cAaveStrategy,
+        signature: "claimGovernance()",
+      },
+      {
+        // Upgrade OUSD implementation
+        contract: cOUSDProxy,
+        signature: "upgradeTo(address)",
+        args: [cOUSDImpl.address],
+      },
+      {
+        // Set VaultCore implementation
+        contract: cVaultProxy,
+        signature: "upgradeTo(address)",
+        args: [cVaultCoreImpl.address],
+      },
+      {
+        // Set VaultAdmin implementation
+        contract: cVault,
+        signature: "setAdminImpl(address)",
+        args: [cVaultAdminImpl.address],
+      },
+      {
+        // Set new trustee
+        contract: cVault,
+        signature: "setTrusteeAddress(address)",
+        args: [cBuyback.address],
+      },
+      {
+        // Collect old OGN
+        contract: cOldBuyback,
+        signature: "transferToken(address,uint256)",
+        args: [cOGN.address, ognBalance],
+      },
+      {
+        // Move old OGN forward to new contract
+        contract: cOGN,
+        signature: "transfer(address,uint256)",
+        args: [cBuyback.address, ognBalance],
+      },
+      {
+        // Set new oracle address
+        contract: cVault,
+        signature: "setPriceProvider(address)",
+        args: [cOracleRouter.address],
+      },
+      {
+        // Add CRV as a swap token
+        contract: cVault,
+        signature: "addSwapToken(address)",
+        args: [assetAddresses.CRV],
+      },
+      {
+        // Add COMP as a swap token
+        contract: cVault,
+        signature: "addSwapToken(address)",
+        args: [assetAddresses.COMP],
+      },
+      {
+        // Add AAVE as a swap token
+        contract: cVault,
+        signature: "addSwapToken(address)",
+        args: [assetAddresses.AAVE],
+      },
+      {
+        // Approve Compound strategy in Vault
+        contract: cVault,
+        signature: "approveStrategy(address)",
+        args: [cCompoundStrategyProxy.address],
+      },
+      {
+        // Remove the deafult strategy for DAI
```

Comment:
@tomlinton This removes the default strategy for DAI, but never adds it back. Should this not be setting the default DAI strategy to Compound?

Reply:
Yea I wasn't sure about that. If I set it to Compound all the DAI will end up in there and I think we were planning to send some to AAVE. I had thought we might manually allocate some to aave, some to compound.

Reply:
The default strategy allowed the invested funds to be withdrawn by users, so if we have most of the funds invested, then we should a default strat.

With all the funds in compound, the strategist just needs to to run a reallocate to move a portion of funds over to AAVE. So just one strategist action, afterwords, rather than two.

Reply:
The vault has a method to reallocate funds from one strategy to another, and we can withdraw certain or all funds from a strategy to the vault, but I don't think we have a method to send funds from the vault to a non-default strategy. In OUSD's normal operation, this is not a probables. Ideally there's sort of a three layer flow going from the vault buffer into and out of a coin's default strategy, and then from the default strategy into and out of other longer term strategies, and reallocate works great for this.

But the only strategy deposit methods in both the VaultCore and VaultAdmin deposit to the default strategy for an asset. This means we will have 9.6 million DAI uninvested until we set a default strategy and allocate.